telegram/ext/_conversationhandler.py

```diff
@@ -283,35 +294,73 @@ def __init__(
         for state_handlers in states.values():
             all_handlers.extend(state_handlers)
 
-        if self.per_message:
-            for handler in all_handlers:
+        # this loop is going to warn the user about handlers which can work unexpected
+        # in conversations
+
+        # this link will be added to all warnings tied to per_* setting
+        per_faq_link = (
+            " Read this FAQ entry to learn more about the per_* settings https://git.io/JtcyU."
+        )
+
+        for handler in all_handlers:
+            if isinstance(handler, (StringCommandHandler, StringRegexHandler)):
+                warn(
+                    "The `ConversationHandler` only handles updates of type `telegram.Update`. "
+                    f"{handler.__class__.__name__} handles updates of type `str`.",
+                    stacklevel=2,
+                )
+            if isinstance(handler, TypeHandler) and handler.type is not Update:
+                warn(
+                    "The `ConversationHandler` only handles updates of type `telegram.Update`."
+                    f" The TypeHandler is set to handle {handler.type.__name__}.",
+                    stacklevel=2,
+                )
+            if isinstance(handler, PollHandler):
+                warn(
+                    "PollHandler will never trigger in a conversation since it has no information "
+                    "about the chat or the user who voted in it. Do you mean the "
+                    "`PollAnswerHandler`?",
+                    stacklevel=2,
+                )
+
+            if (
+                self.per_chat
+                and self.per_user
+                and (
+                    isinstance(
+                        handler,
+                        (
+                            ShippingQueryHandler,
+                            InlineQueryHandler,
+                            ChosenInlineResultHandler,
+                            PreCheckoutQueryHandler,
+                            PollAnswerHandler,
+                        ),
+                    )
+                )
+            ):
+                warn(
+                    f"Updates handled by {handler.__class__.__name__} only have information about "
+                    f"the user, so this handler wont ever be triggered if `per_chat=True`."
+                    + per_faq_link,
+                    stacklevel=2,
+                )
+
+            if self.per_message:
                 if not isinstance(handler, CallbackQueryHandler):
                     warn(
                         "If 'per_message=True', all entry points, state handlers, and fallbacks"
                         " must be 'CallbackQueryHandler', since no other handlers "
-                        "have a message context.",
+                        "have a message context." + per_faq_link,
```

Comment:
```suggestion
                        "have a message context.{per_faq_link}",
```
how come pylint didn't nag about f-string use? Same below and above

Reply:
I think it will only complain about `.format`. though f-strings seem to be [a bit faster](https://stackoverflow.com/a/38722192/10606962). I'm okay with both here …